lib/generators/avo/resource_generator.rb

```diff
@@ -25,6 +25,9 @@ class ResourceGenerator < NamedBaseGenerator
       def create
         return if override_controller?
 
+        # Detect polymorphic associations
+        detect_polymorphic_associations
+
         template "resource/resource.tt", "app/avo/resources/#{resource_name}.rb"
         invoke "avo:controller", [resource_name], options
       end
```

Comment:
I've noticed that all fields are being generated within the `generate_fields` method. Why, then, is `detect_polymorphic_associations` being called separately instead of alongside the other methods inside `generate_fields`? Also, it seems that we're calling it without ever using the return value.